core/domain/user_jobs_one_off.py

```diff
@@ -81,6 +81,64 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class PopulateUserAuthModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating and populating UserAuthModel for
+    all registered users.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        """Marks a job as queued and adds it to a queue for processing."""
+        super(PopulateUserAuthModelOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+
+        # Assumptions:
+        # 1) We are only having full users in the database, and no
+        #    profile users. For profiles we need to do a role check and ensure
+        #    the gae_id attribute is None for them.
+        # 2) This migration works correctly only if gae_id cannot be updated
+        #   in any way for a user. If a gae_id is mapped to some other user by
+        #   deleing it, an error will be raised while migration (see below).
```

Comment:
deleing --> deleting
migration --> migrating

What does "it" refer to? Please be more specific -- I don't understand this last sentence. Also I don't think deleting stuff can cause the gae_id to be remapped -- it is auto-assigned by App Engine.

Reply:
@seanlip Sorry for the poor language.
In the last sentence I wanted to present a scenario - We have done a migration once. There is a user A whose entry exists in the UserAuthModel. Current user A deletes his account (he had a `gae_id` assigned to himself), and new user B signs up with the same google email id - then if there is a possibility that the same `gae_id` is mapped to the new user B also, then while migrating users we will raise an error saying the same `gae_id` is registered to some other user in the UserAuthModel.

My further thinking on it:
1) But, actually if an entry exists for user A, then while deleting user A from UserSettingsModel, we will also mark the corresponding UserAuthModel entry for deletion (this is what the current code will enforce).
2) When a user is deleted, I am not sure what happens with the <gae_id, email id? mapping in the App engine itself. Is it deleted from their records too? Can it happen that a new user comes and sign up with the same gae_id -- this was a question in my mind, but it doesn't really affect me - because of point (1) - UserSettingsModel or UserAuthModel entries would be properly deleted when the user is deleted.

Finally the comment would look like: `This migration works correctly only if gae_id once mapped to a user cannot be updated in any way in future. Currently there is no way to do so, but the migration job would need modifications if it has to work in that setting.`

Reply:
Thanks for the explanation. To answer your question 2, it is the case that that mapping stays the same (it's not maintained within our app).

The final comment looks good. Some wordsmithing: `This migration assumes that, once a gae_id is assigned to a user, it cannot be updated. Currently, there is no way to do so, but the migration job would need modifications if it has to work in such a setting in future.` (Though hopefully we would not need to keep it around that long -- we should delete it as soon as it is run in order to avoid it being run erroneously.)

Reply:
Did the change now.